enterprise/server/remote_execution/commandutil/commandutil_unix.go

```diff
@@ -0,0 +1,132 @@
+//go:build unix
+
+package commandutil
+
+import (
+	"os/exec"
+	"os/user"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+)
+
+type process struct {
+	cmd        *exec.Cmd
+	terminated chan struct{}
+}
+
+func (p *process) preStart() error {
+	return nil
+}
+
+func (p *process) postStart() error {
+	return nil
+}
+
+// killProcessTree kills the given pid as well as any descendant processes.
+//
+// It tries to kill as many processes in the tree as possible. If it encounters
+// an error along the way, it proceeds to kill subsequent pids in the tree. It
+// returns the last error encountered, if any.
+func (p *process) killProcessTree() error {
+	var lastErr error
+
+	// Run a BFS on the process tree to build up a list of processes to kill.
+	// Before listing child processes for each pid, send SIGSTOP to prevent it
+	// from spawning new child processes. Otherwise the child process list has a
+	// chance to become stale if the pid forks a new child just after we list
+	// processes but before we send SIGKILL.
+
+	pidsToExplore := []int{p.cmd.Process.Pid}
+	pidsToKill := []int{}
+	for len(pidsToExplore) > 0 {
+		pid := pidsToExplore[0]
+		pidsToExplore = pidsToExplore[1:]
+		if err := syscall.Kill(pid, syscall.SIGSTOP); err != nil {
+			lastErr = err
+			// If we fail to SIGSTOP, proceed anyway; the more we can clean up,
+			// the better.
+		}
+		pidsToKill = append(pidsToKill, pid)
+
+		childPids, err := ChildPids(pid)
+		if err != nil {
+			lastErr = err
+			continue
+		}
+		pidsToExplore = append(pidsToExplore, childPids...)
+	}
+	for _, pid := range pidsToKill {
+		if err := syscall.Kill(pid, syscall.SIGKILL); err != nil {
+			lastErr = err
+		}
+	}
+
+	return lastErr
+}
+
+// LookupCredential adds credentials to the cmd by resolving a "USER[:GROUP]" string
+// to a credential with both uid and gid populated. Both numeric IDs and non-numeric
+// names can be  specified for either USER or GROUP. If no group is specified, then
+// the user's primary group is used.
+//
+// NOTE: This function does not authenticate that the user is part of the
+// specified group.
+func LookupCredential(cmd *exec.Cmd, spec string) error {
```

Comment:
nit: maybe call it `setCredential` since it's now setting the credential on the cmd instead of returning it?

Reply:
Done.